app/subscriber/src/components/sidebar/hooks/useFilterOptions.ts

```diff
@@ -1,14 +1,52 @@
+import React from 'react';
+import { toast } from 'react-toastify';
 import { useLookup } from 'store/hooks';
 
 export const useFilterOptions = () => {
-  const [{ sources }] = useLookup();
-  const dailyPrint = sources.filter(
-    (source) =>
-      source.code === 'SUN' ||
-      source.code === 'PROVINCE' ||
-      source.code === 'TC' ||
-      source.code === 'GLOBE' ||
-      source.code === 'POST',
-  );
-  return { dailyPrint: dailyPrint, sources };
+  const [{ sources, settings }] = useLookup();
+  const [printIds, setPrintIds] = React.useState<{
+    dailyPrintId: number;
+    weeklyPrintId: number;
+    onlinePrintId: number;
+  }>();
+  const [mediaIds, setMediaIds] = React.useState<{
+    cpWireId: number;
+    talkRadioId: number;
+    newsRadioId: number;
+    televisionId: number;
+  }>();
+
+  React.useEffect(() => {
+    if (settings.length) {
+      if (!settings.find((s) => s.name === 'MediaIds'))
```

Comment:
I wonder if we should change settings to a dictionary so it's simpler to access keys... hmmm

Reply:
I think that is a good idea, would cut out these finds!